crates/precompile/src/cometbft.rs

```diff
@@ -408,7 +408,7 @@ fn encode_light_block_validation_result(
     let mut output =
         vec![0; (VALIDATE_RESULT_METADATA_LENGTH + consensus_state_bytes.len() as u64) as usize];
     output[0] = if validator_set_changed { 1 } else { 0 };
-    output[24..32].copy_from_slice(consensus_state_bytes.len().to_be_bytes().as_ref());
+    output[24..32].copy_from_slice((consensus_state_bytes.len() as u64).to_be_bytes().as_ref());
```

Comment:
Compatibility for 32-bit system, risv32 running panic for this line due to mismatch len